diff --git a/lua/lspconfig/server_configurations/gleam.lua b/lua/lspconfig/server_configurations/gleam.lua
new file mode 100644
index 0000000..620b796
--- /dev/null
+++ b/lua/lspconfig/server_configurations/gleam.lua
@@ -0,0 +1,21 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = {'gleam', 'lsp'},
+    filetypes = { 'gleam' },
+    root_dir = function(fname)
+      return util.root_pattern('gleam.toml', '.git')(fname)
+    end,
+  },
+  docs = {
+    description = [[
+https://github.com/gleam-lang/gleam
+A language server for Gleam Programming Language.
+]],
+    default_config = {
+      cmd = { 'gleam', 'lsp' },
+      root_dir = [[root_pattern("gleam.toml", ".git")]],
+    },
+  },
+}
